analyzers/src/SonarAnalyzer.Common/Rules/VariableUnusedBase.cs

```diff
@@ -43,8 +43,7 @@ protected abstract class UnusedLocalsCollectorBase<TLocalDeclaration>
             public void CollectDeclarations(SyntaxNodeAnalysisContext c) =>
                 declaredLocals.UnionWith(
                     GetDeclaredVariables((TLocalDeclaration)c.Node)
-                        .Select(variable => c.SemanticModel.GetDeclaredSymbol(variable) ?? c.SemanticModel.GetSymbolInfo(variable).Symbol)
-                        .WhereNotNull());
```

Comment:
`WhereNotNull` is defensive and doesn't introduce branching, it's just a method call

Reply:
I discovered it indeed and I reverted the change.
Both of the conditions in `Select` are covered by UTs. I don't know if there's some kind of issue with the `??` operator.